src/main/scala/utils/Connections.scala

```diff
@@ -187,15 +190,17 @@ object Connections {
     }.toMap
 
     val distanceMapAlt: Map[(T, T), Connection[T]] = distanceMap.map {
-      case (tuple, connection) =>
-        (tuple._2, tuple._1) -> connection
+      case ((nodeA, nodeB), connection) =>
+        (nodeB, nodeA) -> connection
     }
 
     Connections(elements, connectionMap.toMap, distanceMap ++ distanceMapAlt)
   }
 
-  /** Method for creating unique [[Connection]]s using a
-    * [[ShortestPathAlgorithm]].
+  /** Method for creating undirected [[Connection]]s using a
+    * [[ShortestPathAlgorithm]]. Excluding connections that have the same source
+    * and target vertex (loops) and duplicated connections.
```

Comment:
Found a better term:

```suggestion
    * and target vertex (loops) and bidirectional connections.
```